assets/pages/regular-contributions/components/formFields.jsx

```diff
@@ -161,29 +168,40 @@ function NameForm(props: PropTypes) {
     <form className={formClassName}>
       {
         !props.isSignedIn ? (
-          <TextInput
-            id="email"
-            value={props.email.value}
-            labelText="Email"
-            placeholder="Email"
-            onChange={props.setEmail}
-            modifierClasses={['email']}
-            showError={shouldShowError(props.email)}
-            errorMessage="Please enter a valid email address."
-            type="email"
-            pattern={emailRegexPattern}
-            required
-          />
+          <div className="email-input">
+            <TextInput
+              id="email"
+              value={props.email.value}
+              labelText="Email"
+              placeholder="Email"
+              onInput={props.setEmail}
+              onChange={props.checkIfEmailHasPassword}
+              modifierClasses={['email']}
+              showError={shouldShowError(props.email, props.checkoutFormHasBeenSubmitted)}
+              errorMessage="Please enter a valid email address."
+              type="email"
+              pattern={emailRegexPattern}
+              required
+            />
+            {/* TODO: style this properly */ }
+            <MustSignIn
+              returnUrl=""
```

Comment:
should we just remove this property? why would we ever want it to not be the current URL?